dashboard/app/controllers/registrations_controller.rb

```diff
@@ -10,7 +10,7 @@ def update
     @user = User.find(current_user.id)
 
     # If email has changed for a non-teacher: clear confirmed_at but don't send notification email
-    if params[:user][:email].present? && !@user.confirmation_required?
+    if  !@user.confirmation_required? && params[:user][:email].present?
```

Comment:
Done (added `params[:user] &&`). Let me know if you prefer the `try` syntax, I consider it slightly less readable when nested.